### PR TITLE
Add set binding

### PR DESCRIPTION
**Pull request by messense**
TODO: 

- [x] frozenset?
- [x] add more tests

Ref #2 

Currently this project is a little hard to get involved because of lacking development guide docs.

### CODE REVIEW

**Comment by messense** [comment]
I am not sure how to implement:

- set iterator
- set union, issuperset, etc. methods.

**Comment by fafhrd91** [comment]
I think `PyIterator` instance is fine

**Comment by fafhrd91** [comment]
lgtm on implementation

do you want to implement `iter()`?

**Comment by messense** [comment]
Switched Travis CI distribution to trusty.

https://blog.travis-ci.com/2017-04-17-precise-EOL

File: src/objects/set.rs [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+pub struct PyFrozenSet(PyObject);
+
+pyobject_newtype!(PySet, PySet_Check, PySet_Type);
+pyobject_newtype!(PyFrozenSet, PyFrozenSet_Check, PyFrozenSet_Type);
+
+impl PySet {
+    /// Creates a new set.
+    ///
+    /// May panic when running out of memory.
+    pub fn new(py: Python, elements: &[PyObject]) -> PySet {
```
Comment (messense) [inline src/objects/set.rs]: What do you think if we change this to:

```rust
pub fn new<T: ToPythonObject>(py: Python: elements: &[T]) -> PySet {}
```

so that we can do:

```rust
let set = PySet::new(py, &[1, 2, 3]);
```

instead of

```rust
let set = PySet::new(py, &[1.to_py_object(py).into_object()]);
```

Comment (fafhrd91) [inline src/objects/set.rs]: I think, this is very good idea

Comment (messense) [inline src/objects/set.rs]: But other things like `PyList` still uses `pub fn new(py: Python, elements: &[PyObject]) -> PyList`, I guess we should do this in a separate PR to do them all.

Comment (fafhrd91) [inline src/objects/set.rs]: I think you should have write access to repo. this kind of change you can just commit.

Comment (messense) [inline src/objects/set.rs]: OK, I'll merge this then do other things.
